app/src/main/java/com/mstudio/superheroarch/MainActivity.kt

```diff
@@ -31,28 +33,35 @@ class MainActivity : AppCompatActivity() {
             v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
             insets
         }
-        val button = findViewById<Button>(R.id.button_to_change_text)
-        characterNameTitle = findViewById(R.id.character_name)
-        characterStatusTitle = findViewById(R.id.character_status)
+        updateButton = findViewById(R.id.refresh_button)
+
+        val characterListRecyclerView: RecyclerView = findViewById(R.id.character_list)
+        characterListRecyclerView.layoutManager = LinearLayoutManager(this)
+        characterListAdapter = CharacterListAdapter(mutableListOf())
+        characterListRecyclerView.adapter = characterListAdapter
+
+        getDataFromRemote()
 
-        button.setOnClickListener {
+        updateButton.setOnClickListener {
             getDataFromRemote()
         }
     }
 
     private fun getDataFromRemote() {
+        val listOfCharacters = mutableListOf<Character>()
         val apiService = RetrofitInstance.getInstance().create(RickAndMortyApi::class.java)
         val call = apiService.doGetCharacters()
         call.enqueue(object : Callback<CharactersResponse> {
             override fun onResponse(call: Call<CharactersResponse>, response: Response<CharactersResponse>) {
                 if (response.isSuccessful) {
-                    val firstCharacter = response.body()?.results?.firstOrNull()
-                    firstCharacter?.let {
-                        characterNameTitle?.text = getString(R.string.character_name, it.name)
-                        characterStatusTitle?.text = getString(R.string.character_status, it.status)
-                        Picasso.get().load(it.image).into(findViewById<ImageView>(R.id.character_image));
-                    } ?: Snackbar.make(findViewById(R.id.main), getString(R.string.error_message_no_character_info), Snackbar.LENGTH_SHORT).show()
+                    response.body()?.results?.let {
+                        for (character in response.body()?.results ?: emptyList()) {
```

Comment:
you don't need this for. Maybe is a mistake due to the naming. The val "result" is characterList. With that name you'll see easily that you just need to do response.body()?.characterList?.let { listOfCharacters.add(it) }